fix: BROS-109: Import Dropzone content area moved beyond fold if JSON preview is too large

ricardoantoniocm opened the pull request:
**Reason for change**

The content of the import dropzone area was moving off-screen when the JSON preview was too large, hindering user experience. This change ensures that the import dropzone content remains visible regardless of the JSON preview's size.

**Screenshots**

**Before**
<img width="1290" alt="before-light" src="https://github.com/user-attachments/assets/e3d0b3ce-124e-49e9-88c3-81ee2caa3996" />

**After**
<img width="1289" alt="after-light" src="https://github.com/user-attachments/assets/27baac66-270c-4c7a-958f-c3f37cd42610" />

**Testing**

The following tests were performed to verify the changes:

  * **Manual UI Verification:** Tested the Project Import page with various JSON preview sizes (small, medium, large) to ensure the import dropzone content remained visible and correctly positioned.
  * **Component Interaction Testing:** Verified that all interactive elements within the import dropzone and JSON preview areas function as expected (e.g., file drag and drop, sample data selection, error message display).

**Reviewer notes**
  * Pay close attention to the visual alignment and responsiveness of the import dropzone and JSON preview areas, especially with large JSON payloads.
  * Verify the application of the new custom scrollbar styling and its impact on user experience.
  * Confirm that the `CodeBlock` and `SimpleCard` component improvements (e.g., `className` prop, flush props) function as intended and provide the expected styling flexibility.

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

ricardoantoniocm committed:
fix: BROS-109: Import dropzone content area moved beyond fold if JSON preview is too large

ricardoantoniocm committed:
Adds support to inject classes for the simple card header and content, as well as to set the header and content elements flush with the card's border.

ricardoantoniocm committed:
Adds support to pass class names to code block component.

ricardoantoniocm committed:
Prevents large JSON samples from pushing the Import dropzone beyond the fold. Improves style of code block.

ricardoantoniocm committed:
Removes unnecessary code.

ricardoantoniocm commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15786970999)

ricardoantoniocm committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15786970999

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-bros-109/import-dropzone-content-area-moved-beyond-fold-if-json-preview-too-large'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15786970999

ricardoantoniocm commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15825131204)

ricardoantoniocm committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15825131204

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-bros-109/import-dropzone-content-area-moved-beyond-fold-if-json-preview-too-large'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15825131204

nass600 commented on web/apps/labelstudio/src/pages/CreateProject/Import/Import.jsx:
```diff
@@ -493,19 +493,28 @@ export const ImportPage = ({
               <div className="w-full h-full flex flex-col min-h-[400px]">
                 {projectConfigured ? (
-                  <SimpleCard title="Expected input preview" className="w-full h-full">
+                  <SimpleCard
```
Probably not part of this PR but it seems we have Card, CardNew and SimpleCard as Design System components, none of them included in Storybook

ricardoantoniocm replied:
Indeed! We should work to consolidate these. SimpleCard is new, from when the Home page was implemented. I also think we should avoid duplicating existing components unless there is an extremely good reason.

ricardoantoniocm commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15863168211)

ricardoantoniocm committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15863168211

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-bros-109/import-dropzone-content-area-moved-beyond-fold-if-json-preview-too-large'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15863168211

ricardoantoniocm commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15877926205)
> Successfully merged:  4 files changed, 51 insertions(+), 274 deletions(-)

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-bros-109/import-dropzone-content-area-moved-beyond-fold-if-json-preview-too-large'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15877926205
